dojo/reimport_utils.py

```diff
@@ -0,0 +1,58 @@
+from django.conf import settings
+from dojo.models import Finding
+
+import logging
+
+logger = logging.getLogger(__name__)
+deduplicationLogger = logging.getLogger("dojo.specific-loggers.deduplication")
+
+"""
+Common code for reimporting from APIV2 or from the GUI
+"""
+
+
+def get_deduplication_algorithm_from_conf(scan_type):
+    # Default algorithm
+    deduplication_algorithm = 'legacy'
+    # Check for an override for this scan_type in the deduplication configuration
+    if hasattr(settings, 'DEDUPLICATION_ALGORITHM_PER_PARSER') and scan_type in settings.DEDUPLICATION_ALGORITHM_PER_PARSER:
+        deduplication_algorithm = settings.DEDUPLICATION_ALGORITHM_PER_PARSER[scan_type]
+    return deduplication_algorithm
+
+
+def match_new_finding_to_existing_finding(item, test, deduplication_algorithm, scan_type):
+    if deduplication_algorithm == 'hash_code':
+        return Finding.objects.filter(
+            test=test,
+            hash_code=item.hash_code).exclude(
+                        hash_code=None).all()
+    elif deduplication_algorithm == 'unique_id_from_tool' or deduplication_algorithm == 'unique_id_from_tool_or_hash_code':
+        # processing 'unique_id_from_tool_or_hash_code' as 'unique_id_from_tool' because when using 'unique_id_from_tool_or_hash_code'
+        # we usually want to use 'hash_code' for cross-parser matching, 
+        # while it makes more sense to use the 'unique_id_from_tool' for same-parser matching
+        return Finding.objects.filter(
+            test=test,
+            unique_id_from_tool=item.unique_id_from_tool).exclude(
+                        unique_id_from_tool=None).all()
```

Comment:
should you not include the findings with the same hash code if there is no match on unique_id_from_tool?

Reply:
i think i'd go for either: 
1. only unique_id_from_tool (because as explained in the comments, for the use cases I had in mind, when this configuration is on, it doesn't make much sense to match same parser findings on hash_code)
2. both unique_id_from_tool and hash_code (more coherent with the actual configuration)

2# is probably more logical

But matching on hash_code only if the match on unique_id_from_tool doesn't find anything seems a bit unpredictible. Although I don't have a strong opinion on this right now, need to throw the code at more tests.

Reply:
@valentijnscholten i'll go for 2: both unique_id_from_tool and hash_code because this is how it's done when deduplicating outside of re-import, so better to keep the same logic. 

will push that asap